examples/backward_modes.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python3
+#
+# This example illustrates the three backward modes (FULL, IMPLICIT, and TRUNCATED)
+# on a problem fitting a quadratic to data.
+#
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import theseus as th
+import theseus.optimizer.nonlinear as thnl
```

Comment:
Should also add `BackwardMode` to `theseus/__init__.py` so it's available in `import theseus as th` directly.

Reply:
Added, and updated the example/test to use this from `th` directly rather than `thnl`